java-viewer/src/main/java/org/sonar/java/viewer/Viewer.java

```diff
@@ -83,20 +95,32 @@ private static String generate(Request request) {
   private static String renderIndex(String javaCode) {
     HashMap<String, String> values = new HashMap<>();
 
-    CFG cfg = CFGViewer.buildCFG(javaCode);
+    CompilationUnitTree cut = (CompilationUnitTree) PARSER.parse(javaCode);
+    SemanticModel semanticModel = SemanticModel.createFor(cut, new SquidClassLoader(Collections.emptyList()));
+    MethodTree firstMethod = getFirstMethod(cut);
+
+    CFG cfg = CFG.build(firstMethod);
 
     values.put("cfg", CFGDebug.toString(cfg));
 
-    values.put("dotAST", ASTViewer.toDot(javaCode));
-    values.put("dotCFG", CFGViewer.toDot(cfg));
-    values.put("dotEG", EGViewer.toDot(javaCode, cfg.blocks().get(0).id()));
+    values.put("dotAST", new ASTDotGraph(cut).toDot());
+    values.put("dotCFG", new CFGDotGraph(cfg).toDot());
+    values.put("dotEG", new EGDotGraph(cut, firstMethod, semanticModel, cfg.blocks().get(0).id()).toDot());
 
     values.put("errorMessage", "");
     values.put("errorStackTrace", "");
 
     return renderWithValues(javaCode, values);
   }
 
+  private static MethodTree getFirstMethod(CompilationUnitTree cut) {
+    ClassTree classTree = (ClassTree) cut.types().get(0);
+    return (MethodTree) classTree.members().stream()
+      .filter(m -> m.is(Tree.Kind.METHOD))
+      .findFirst()
+      .orElse(null);
```

Comment:
given that this method is not annotated and the null value is passed to CFG.build without null check,  this means this is not unit tested AND this should problaby be  null checked to return a nice error.